app/controllers/api/v1/targets_controller.rb

```diff
@@ -11,7 +11,23 @@ def index
       def create
         @target = current_user.targets.build(target_params)
         if @target.save
-          @target
+          match = search_match(@target.radius, @target.latitude, @target.longitude, @target.topic_id)
+          matched_user = []
+          match_conversation = []
+          
+          if match.size > 0
```

Comment:
This create methods is getting too long. There is a good practice that says: keep controller methods skinny. You can move this logic to a create target service.

This comment is not mandatory but to keep it in mind. Feel free to ignore me 😁